src/test/ui/issues/issue-78957.stderr

```diff
@@ -0,0 +1,29 @@
+error[E0518]: attribute should be applied to function or closure
+  --> $DIR/issue-78957.rs:4:16
+   |
+LL | pub struct Foo<#[inline] const N: usize>;
+   |                ^^^^^^^^^       - not a function or closure
+
+error: attribute should be applied to a function
+  --> $DIR/issue-78957.rs:6:16
+   |
+LL | pub struct Bar<#[cold] const N: usize>;
+   |                ^^^^^^^       - not a function
+   |
+note: the lint level is defined here
+  --> $DIR/issue-78957.rs:1:9
+   |
+LL | #![deny(unused_attributes)]
+   |         ^^^^^^^^^^^^^^^^^
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
```

Comment:
So it seems like we always just emit a future compat lint if `cold` is applied incorrectly :thinking: 

It would be nice to increase that lint to `deny` by default. Don't know how much this would break though

Reply:
I'll leave this for a follow-up.